ethtxmanager/sqlstorage/sqlstorage.go

```diff
@@ -0,0 +1,479 @@
+package sqlstorage
+
+import (
+	"context"
+	"database/sql"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"time"
+
+	localCommon "github.com/0xPolygon/zkevm-ethtx-manager/common"
+	"github.com/0xPolygon/zkevm-ethtx-manager/log"
+	"github.com/0xPolygon/zkevm-ethtx-manager/types"
+	"github.com/ethereum/go-ethereum/common"
+	_ "github.com/mattn/go-sqlite3"
+	migrate "github.com/rubenv/sql-migrate"
+)
+
+const (
+	// Insert denotes insert statement
+	Insert sqlAction = iota
+	// Update denotes update statement
+	Update
+	// Delete denotes delete statement
+	Delete
+)
+
+type sqlAction int
+
+func (s sqlAction) String() string {
+	switch s {
+	case Insert:
+		return "INSERT"
+
+	case Update:
+		return "UPDATE"
+
+	case Delete:
+		return "DELETE"
+	}
+
+	return "UNKNOWN"
+}
+
+const (
+	// baseSelectQuery represents the base select query, that retrieves all the values from the monitored_txs table
+	baseSelectQuery = `SELECT id, from_address, to_address, nonce, value, tx_data, gas, gas_offset, gas_price, 
+							blob_sidecar, blob_gas, blob_gas_price, gas_tip_cap, status, 
+							block_number, history, created_at, updated_at, estimate_gas
+						FROM monitored_txs`
+
+	// baseDeleteStatement represents the base delete statement that deletes all the records from the monitored_txs table
+	baseDeleteStatement = "DELETE FROM monitored_txs"
+)
+
+var _ types.StorageInterface = (*SqlStorage)(nil)
+
+// SqlStorage encapsulates logic for MonitoredTx CRUD operations.
+type SqlStorage struct {
+	db *sql.DB
+}
+
+// NewStorage creates and returns a new instance of SqlStorage with the given database path.
+// It first opens a connection to the SQLite database and then runs the necessary migrations.
+// If any error occurs during the database connection or migration process, it returns an error.
+func NewStorage(driverName, dbPath string) (*SqlStorage, error) {
+	db, err := sql.Open(driverName, dbPath)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := RunMigrations(driverName, db, migrate.Up); err != nil {
+		return nil, err
+	}
+
+	return &SqlStorage{db: db}, nil
+}
+
+// Add persist a monitored transaction into the SQL database.
+func (s *SqlStorage) Add(ctx context.Context, mTx types.MonitoredTx) error {
+	mTx.CreatedAt = time.Now()
+	mTx.UpdatedAt = mTx.CreatedAt
+
+	query := `
```

Comment:
Maybe take a look at the `meddler` we used on `cdk` repo, to handle inserting and querying structs from sqlite?